File: bintree_csharp/binarytreedisp/ValueUtil.cs
using System;


namespace my.binarytreedisp
{
    class ValueUtil
    {
        private int _FloatPrecision;
        private string FormatString;


        public ValueUtil(int floatPrecision = 2)
        {
            this.FloatPrecision = floatPrecision;
        }



        public int FloatPrecision
        {
            get
            {
                return _FloatPrecision;
            }

            set
            {
                if (value < 0)
                    throw new ArgumentException("FloatPrecision must be a non-negative integer");

                this._FloatPrecision = value;
                this.FormatString = "0." + new string('#', value);
            }
        }



        public string GetStr<T>(T value)
        {
            if (value is null)
                return string.Empty;

            if (value is float valueF) return valueF.ToString(this.FormatString);
            if (value is double valueD) return valueD.ToString(this.FormatString);
            if (value is decimal valueM) return valueM.ToString(this.FormatString);

            return value.ToString();
        }
    }
}

File: bintree_py/binarytreedisp/valueutil.py
class ValueUtil:
    #
    #
    def __init__(self, float_precision: int = 2):
        self.set_float_precision(float_precision)

    #
    #
    def get_float_precision(self):
        return self.__float_precision

    #
    #
    def set_float_precision(self, precision: int):
        if type(precision) is not int or precision < 0:
            raise ValueError('Invalid argument: float precision must be a non-negative integer')

        self.__float_precision = precision

    #
    #
    def get_str(self, value) -> str:
        type_value = type(value)

        if type_value is float:
            return str(round(value, self.__float_precision))

        if type_value is str:
            return value

        return str(value)
